sls/test/src/org/scala/abusers/sls/integration/LSPIntegrationTestSuite.scala

```diff
@@ -0,0 +1,180 @@
+package org.scala.abusers.sls.integration
+
+import cats.effect.kernel.Resource
+import cats.effect.IO
+import cats.syntax.all.*
+import fs2.io.file.Path
+import lsp.*
+import org.scala.abusers.pc.PresentationCompilerProvider
+import org.scala.abusers.sls.*
+import org.scala.abusers.sls.integration.utils.TestLSPClient
+import org.scala.abusers.sls.integration.utils.TestWorkspace
+import weaver.SimpleIOSuite
+
+import java.net.URI
+
+abstract class LSPIntegrationTestSuite extends SimpleIOSuite {
+
+  case class TestServerContext(
+      server: ServerImpl,
+      client: TestLSPClient,
+      workspace: TestWorkspace,
+  )
+
+  protected def withServer(workspace: Resource[IO, TestWorkspace]): Resource[IO, TestServerContext] =
+    for {
+      workspace <- workspace
+      client    <- TestLSPClient.create.toResource
+      serverCtx <- createServer(client, workspace)
+    } yield TestServerContext(serverCtx, client, workspace)
+
+  protected def withSimpleServer: Resource[IO, TestServerContext] =
+    withServer(TestWorkspace.withSimpleScalaProject)
+
+  protected def withMultiModuleServer: Resource[IO, TestServerContext] =
+    withServer(TestWorkspace.withMultiModuleProject)
+
+  protected def withMockBSPServer(workspace: Resource[IO, TestWorkspace]): Resource[IO, TestServerContext] =
+    for {
+      workspace <- workspace
+      client    <- TestLSPClient.create.toResource
+      mockBSP   <- org.scala.abusers.sls.integration.bsp.utils.MockBSPServer.withDefaultTargets.toResource
+      serverCtx <- createServerWithMockBSP(client, workspace, mockBSP)
+    } yield TestServerContext(serverCtx, client, workspace)
+
+  protected def withLanguageFeaturesServer: Resource[IO, TestServerContext] =
+    withMockBSPServer(TestWorkspace.withSimpleScalaProject)
+
+  private def createServer(client: TestLSPClient, workspace: TestWorkspace): Resource[IO, ServerImpl] =
```

Comment:
issue: `ServerImpl` shouldn't be used in methods like this (if you ask me, the class itself should be private, and the return type of `ServerImpl.create` should be `SlsLanguageServer[IO]`)